src/nvim/api/vim.c

```diff
@@ -695,6 +696,73 @@ Dictionary nvim_get_color_map(void)
   return colors;
 }
 
+// Helper function for nvim_get_keymap and nvim_buf_get_keymap
+// @param mode The abbreviation for the mode
+// @param buffer_id
+//      -1: Global mapping
+//      0: Current buffer
+//      else: The desired buffer ID
+//
+// @returns An array of maparg() like dictionaries describing mappings
+static Array get_keymap_helper(String mode, Integer buffer_id)
```

Comment:
Avoid naming anything "helper". All functions are helpful, so it doesn't add information. 

Just `keymap_array()` is an OK name for this, analogous to `cursor_shape_dict()`.